script.js

```diff
@@ -1,4 +1,208 @@
+var game_state_select_num_of_dice = "player select num of dice";
+var game_state_roll_dice = "player roll your dice";
+var game_state_autogenerate_order = "player computer autogenerates order";
+var game_state_find_winner = "find winner";
+var currentGameState = game_state_select_num_of_players;
+
+var currentPlayer = 0;
+var game_mode_setting = "select normal or reverse";
+var game_mode_normal = "highest";
+var game_mode_reverse = "lowest";
+var currentGameMode = game_mode_setting;
+
+//define global variables
+var diceLists = [];
+var diceSelected = "";
+var userName = "";
+var totalPlayers = 0;
+var playerList = [];
+
+var numOfDiceSelected = function (num) {
+  var numOfDice = num;
+  if (num != Number(num)) {
+    return -1;
+  } else {
+    return Number(numOfDice);
+  }
+};
+
+var diceRollList = function (numOfDice) {
+  var randomDiceNum = function () {
+    return Math.floor(Math.random() * 6 + 1);
+  };
+  var diceList = [];
+  index = 0;
+  while (index < numOfDice) {
+    diceList.push(randomDiceNum());
+    index = index + 1;
+  }
+  return diceList;
+};
+
+var autoGeneratedList = function (diceList, mode) {
+  var sortedDiceList;
+  if (mode == "highest") {
+    sortedDiceList = diceList.sort(function (a, b) {
+      return a - b;
+    });
+  } else if (mode == "lowest") {
+    sortedDiceList = diceList.sort(function (a, b) {
+      return b - a;
+    });
+  }
+  return sortedDiceList;
+};
+
+var autoGeneratedNum = function (sortedDiceList) {
+  var sortedListStrings = [];
+  index = 0;
+  while (index < sortedDiceList.length) {
+    sortedListStrings.push(sortedDiceList[index].toString());
+    index = index + 1;
+  }
+  var totalNumber = "";
+  index = 0;
+  while (index < sortedListStrings.length) {
+    var element = sortedListStrings[index];
+    totalNumber = element + totalNumber;
+    index = index + 1;
+  }
+  return Number(totalNumber);
+};
+
+var winningOutcome = function (mode, diceLists) {
+  var currentWinnerIndex;
+  var currentWinningScore;
+  if (mode == "highest") {
+    var index = 0;
+    while (index < diceLists.length) {
+      var playerScore = autoGeneratedNum(diceLists[index]);
+      if (
+        playerScore > currentWinningScore ||
+        currentWinningScore == undefined
+      ) {
+        currentWinningScore = playerScore;
+        currentWinnerIndex = index;
+      }
+      index = index + 1;
+    }
+  }
+  if (mode == "lowest") {
+    var index = 0;
+    while (index < diceLists.length) {
+      var playerScore = autoGeneratedNum(diceLists[index]);
+      if (
+        playerScore < currentWinningScore ||
+        currentWinningScore == undefined
+      ) {
+        currentWinningScore = playerScore;
+        currentWinnerIndex = index;
+      }
+      index = index + 1;
+    }
+  }
+  console.log(
+    "current winner index",
+    currentWinnerIndex,
+    currentWinningScore,
+    mode
+  );
+  return [currentWinnerIndex, currentWinningScore, mode];
```

Comment:
great job here! really like the use of an array to return 3 values from a function!!